rholang/src/main/k/rholang/tests/Sending-Receiving/logical-and-within-patterns.rho

```diff
@@ -0,0 +1,24 @@
+// The following program consists of a receive which listens for a pattern
+// which itself is a send over *either* @1 or @2. It grabs whatever is being
+// sent and sends it over @Nil.
+
+// This will nondeterministically either rewrite to:
+//     @Nil!(@1!("first succeeds"))
+//     @Nil!("second succeeds")
+// or
+//     @Nil!(@2!("second succeeds"))
+//     @Nil!("first succeeds")
```

Comment:
It will be either
```
@Nil!("first succeeds")
@Nil!(@"second succeeds"!(Nil))
```

or

```
@Nil!("second succeeds")
@Nil!(@"first succeeds"!(Nil))
```

no **@1! …** or **@2! …**